### PR TITLE
4 get/api/articles/article

### PR DESCRIPTION
**Pull request by Phil-Code**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -42,4 +41,39 @@ describe('GET/api', ()=>{
...
+        .then(({body})=>{
+            const article= body.article
+            expect(article.title).toBe('Sony Vaio; or, The Laptop')
+            expect(article.topic).toBe('mitch')
+            expect(article.author).toBe('icellusedkars')
+            expect(typeof article.body).toBe('string')
+            expect(typeof article.created_at).toBe('string')
+            expect(typeof article.votes).toBe('number')
+            expect(article.article_img_url).toBe('https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700')
+        })
```
Comment (NRMSMITH) [inline __tests__/endpoints.test.js:65]: For a more flexible test suite - I would suggest looking at the .toMatchObject or .objectContaining matchers :)

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -42,4 +41,39 @@ describe('GET/api', ()=>{
...
+    })
+    test('returns 400 status code and bad request message when client enters an invalid id type', ()=>{
+        return request(app)
+        .get('/api/articles/invalid_input')
+        .expect(400)
+        .then(({body})=>{
+            expect(body.msg).toBe('invalid id type')
+        })
+    })
```
Comment (NRMSMITH) [inline __tests__/endpoints.test.js:83]: 👍 good error testing

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const { fetchArticleById } = require("../models/articles.models");
+
+exports.getArticleById = async (req, res, next) =>{
+    const id = req.params.article_id
+
+    try {
+        const article = await fetchArticleById(id);
+        res.status(200).send({article})
+    } catch(err){
+        if (err.code === '22P02'){
```
Comment (NRMSMITH) [inline controllers/articles.controllers.js]: This really should be in an error handler, not on a specific endpoint controller (you'll have to duplicate it a lot then you see)

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
-A change to your test
-Error handling middleware needed in app.js or in a errors.controllers.js

**Review by B7Bprog — APPROVED** [review]
Hey Phil, nice refactoring, please merge!

**Pull request merged**